pkg/controller/service/service_controller.go

```diff
@@ -211,10 +215,13 @@ func (s *ServiceController) Run(stopCh <-chan struct{}, workers int) {
 	defer runtime.HandleCrash()
 	defer s.queue.ShutDown()
 
-	klog.Info("Starting service controller")
+	klog.Infof("Starting service controller with node lister #= %d", len(s.nodeListers))
 	defer klog.Info("Shutting down service controller")
 
-	if !controller.WaitForCacheSync("service", stopCh, s.serviceListerSynced, s.nodeListerSynced) {
+	if !controller.WaitForCacheSync("service (w/o node)", stopCh, s.serviceListerSynced) {
+		return
+	}
+	if !nodeutil.WaitForNodeCacheSync("service (node)", s.nodeListersSynced) {
```

Comment:
is there a reason WaitForNodeCacheSync() not to have stopCh param as WaitForCacheSync()?

Reply:
WaitForCacheSync() is generic util, WaitForNodeCacheSync is customized for node informer sync only. They have different implementation. Added a note to consider unify their implementations.